src/components/Launcher.js

```diff
@@ -24,6 +24,7 @@ class Launcher extends Component {
   }
 
   playIncomingMessageSound() {
+    if (this.props.mute) { return; }
```

Comment:
I would say `playIncomingMessageSound` should reliably `play the incoming message` sound and the if-statement should decide wether or not this function is called in the first place